ocf_data_sampler/config/model.py

```diff
@@ -0,0 +1,236 @@
+""" Configuration model for the dataset.
+
+All paths must include the protocol prefix.  For local files,
+it's sufficient to just start with a '/'.  For aws, start with 's3://',
+for gcp start with 'gs://'.
+
+"""
+
+import logging
+from typing import Dict, List, Optional, Union
+
+from pydantic import BaseModel, Field, RootModel, field_validator, ValidationInfo
+
+from ocf_datapipes.utils.consts import NWP_PROVIDERS
+
+logger = logging.getLogger(__name__)
+
+providers = ["pvoutput.org", "solar_sheffield_passiv"]
+
+
+class Base(BaseModel):
+    """Pydantic Base model where no extras can be added"""
+
+    class Config:
+        """config class"""
+
+        extra = "forbid"  # forbid use of extra kwargs
+
+
+class General(Base):
+    """General pydantic model"""
+
+    name: str = Field("example", description="The name of this configuration file.")
+    description: str = Field(
+        "example configuration", description="Description of this configuration file"
+    )
+
+
+class DataSourceMixin(Base):
+    """Mixin class, to add forecast and history minutes"""
+
+    forecast_minutes: int = Field(
+        None,
```

Comment:
I think this should be a required field and so should be `...` instead of `None`?

Reply:
I think it's optional because of satellite? It's allowed to not have forecast, but yeah we can make it required and set to 0

Reply:
I see what you mean. I think that since it is required for some sources it should still be required here. I have been generally setting it to zero for the satellite inputs. Hopefully this should be cleaner once we switch to `interval_[start/end]`